click_odoo_contrib/update.py

```diff
@@ -97,6 +184,26 @@ def OdooEnvironmentWithUpdate(database, ctx, **kwargs):
     Registry.new(database, update_module=True)
     with conn.cursor() as cr:
         _save_installed_checksums(cr)
+
+
+@contextmanager
+def OdooEnvironmentWithUpdate(database, ctx, **kwargs):
+    # Watch for database locks while Odoo updates
+    watcher_process = multiprocessing.Process(
+        target=_db_lock_watcher,
+        args=(database, ctx.params["watcher_beat"], ctx.params["watcher_max_seconds"]),
+    )
+    try:
+        if ctx.params["watcher_max_seconds"] > 0:
+            watcher_process.start()
+        # Update Odoo datatabase
+        _update_db(database, ctx.params["update_all"], ctx.params["i18n_overwrite"])
```

Comment:
The update process is moved to its own method, so code in this method looks more clean because all lit does it to handle both the update and the watcher methods..